js/binance.js

```diff
@@ -1357,17 +1385,51 @@ module.exports = class binance extends Exchange {
         return result;
     }
 
+    market (symbol) {
+        // helper method for legacy symbols
+        // this means that 'BTC/USDT' will continue to work for futures
+        if (symbol in this.markets) {
+            // 99% of calls should take this path and would not be slowed down
+            return this.markets[symbol];
+        } else {
+            const defaultType = this.getDefaultType ();
+            if (defaultType === 'linear') {
+                if (symbol.indexOf (':') === -1) {
+                    const parts = symbol.split ('/');
+                    const quote = parts[1];
+                    symbol = symbol + ':' + quote;
+                }
+            } else if (defaultType === 'inverse') {
+                if (symbol.indexOf (':') === -1) {
+                    const parts = symbol.split ('/');
+                    const base = parts[1];
+                    symbol = symbol + ':' + base;
+                }
+            }
+            return super.market (symbol);
+        }
+    }
+
     async fetchMarkets (params = {}) {
-        const defaultType = this.safeString2 (this.options, 'fetchMarkets', 'defaultType', 'spot');
-        const type = this.safeString (params, 'type', defaultType);
+        const types = this.safeValue (this.options, 'fetchMarkets', [ 'spot', 'linear', 'inverse' ]);
+        let result = [];
+        for (let i = 0; i < types.length; i++) {
+            const type = types[i];
+            const response = await this.fetchMarketsByType (type, params);
+            result = this.arrayConcat (result, response);
+        }
+        return result;
+    }
+
+    async fetchMarketsByType (type, params = {}) {
         const query = this.omit (params, 'type');
-        if ((type !== 'spot') && (type !== 'future') && (type !== 'margin') && (type !== 'delivery')) {
-            throw new ExchangeError (this.id + " does not support '" + type + "' type, set exchange.options['defaultType'] to 'spot', 'margin', 'delivery' or 'future'"); // eslint-disable-line quotes
+        if ((type !== 'spot') && (type !== 'linear') && (type !== 'inverse')) {
+            throw new ExchangeError (this.id + " does not support '" + type + "' type, set exchange.options['defaultType'] to 'spot', 'margin', 'inverse' or 'linear'"); // eslint-disable-line quotes
```

Comment:
Since `margin` markets are loaded alongside spot markets, I believe we should either remove it from the valid "types" in this error message or add it to the if condition above. Currently the error message misleads the user a bit